pkg/azure/api/api.go

```diff
@@ -351,6 +351,8 @@ func parseSubnet(subnet *network.Subnet) (s *ipamTypes.Subnet) {
 		s.CIDR = c
 		if subnet.IPConfigurations != nil {
 			s.AvailableAddresses = c.AvailableIPs() - len(*subnet.IPConfigurations)
+		} else {
+			s.AvailableAddresses = c.AvailableIPs()
```

Comment:
The change makes sense, maybe add a comment with a not explaining the motivation for future reference?

Reply:
I'm pretty sure this case will happen only if there are too many ipconfigs for ARM to return in the response. Is there any negative impact from overestimating the number of available IPs?

Reply:
Added a comment about this.